src/common/btree.c

```diff
@@ -916,6 +917,62 @@ btr_root_grow(struct btr_context *tcx, TMMID(struct btr_node) mmid_left,
 	return 0;
 }
 
+struct btr_check_vbt {
+	TMMID(struct btr_node)	nd_mmid;
+	int			at;
+	uint32_t		intent;
```

Comment:
Not a big deal, but I think it's probably better to simplify the intent to "SKIP_UNCOMMITTED", "CLEAR_UNCOMMITED" or DEFAULT? and looks the intent can be merged into probe_opc, so that probe operation not only specify the position, it also specify if the 2PC non-committed leaf needs be killed or skipped.

Reply:
From the caller view, it should not know too much about the DTX detail, such as the status. It is tree API inside logic to decide whether some record should be returned the caller according the DTX status + the caller given intention. What the caller knows is that it wants to do, that is the operation intention. We can combine the intent with probe_obc, we did that before. But the shorting-coming is that: some callers may miss to set the intent when call probe(), it will cause subtle trouble. That is why I make the @intent as separated parameter.